src/ndarray/ndarray.cc

```diff
@@ -440,13 +440,13 @@ void NDArray::Chunk::SetMKLMem(const TShape &shape, int dtype) {
   }
   mkldnn::memory::desc data_md{dims, get_mkldnn_type(dtype), layout};
   auto cpu_engine = CpuEngine::Get()->get_engine();
-  if (shandle.dptr == nullptr) {
+  if (shandle.GetDptr() == nullptr) {
```

Comment:
Since you have some concern about this line, you should add some comment like "FIXME(sotskin): GetDptr() never returns a nullptr."